zk/stages/stage_batches.go

```diff
@@ -304,7 +304,7 @@ LOOP:
 
 			// check for sequential block numbers
 			if l2Block.L2BlockNumber != lastBlockHeight+1 {
-				return fmt.Errorf("block number is not sequential, expected %d, got %d", lastBlockHeight+1, l2Block.L2BlockNumber)
+				panic(fmt.Sprintf("block number is not sequential, restart required, expected %d, got %d", lastBlockHeight+1, l2Block.L2BlockNumber))
```

Comment:
instead of a panic here, could we skip the entry and log that we've done it?  We might end up in a scenario where we have a bug that duplicates something in the stream and this in theory would be fine to just skip rather than panic?

Reply:
but the issue here that sometimes it just not the right number due to unwinds -- so restart fixes it

Reply:
ahhhh, so a sticky plaster for now then 👍  all good in that case

Reply:
yeah yeah